extension/goext/database.go

```diff
@@ -67,13 +67,25 @@ func withinJoinable(tx ITransaction, fn func(tx ITransaction) error) error {
 	return fn(tx)
 }
 
+func closeTx(tx ITransaction) {
+	err := tx.Close()
+	if err != nil {
+		log.Warning(fmt.Sprintf("close scoped database transaction failed with error: %s", err))
+	}
+}
+
 func withinDetached(db IDatabase, context Context, txBegin func() (ITransaction, error), fn func(tx ITransaction) error) error {
 	opts := db.Options()
 	retryTxCount := opts.RetryTxCount
 	retryTxInterval := opts.RetryTxInterval
 	var tx ITransaction
 	var err error
 
+	defer func() {
+		closeTx(tx)
```

Comment:
for safety, `tx.Closed` should be checked?

Reply:
@morrisson 
I've added check in closeTx.